### PR TITLE
Add cloud pods collaborative debugging tutorial

### PR DESCRIPTION
**Pull request by tinyg210**
Add tutorial for using cloud pods for collaborative development and debugging in teams

### CODE REVIEW

File: content/en/tutorials/cloud-pods-collaborative-debugging/index.md [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+Options:
+  -h, --help  Show this message and exit.
+
+Commands:
+  delete    Delete a Cloud Pod
+  list      List all available Cloud Pods
+  load      Load the state of a Cloud Pod into the application runtime/ Users can import Cloud Pods from...
+  remote    Manage cloud pod remotes
+  save      Create a new Cloud Pod
+  versions  List all available versions for a Cloud Pod
```
Comment (giograno) [inline content/en/tutorials/cloud-pods-collaborative-debugging/index.md:227]: `for` is colored differently. Any idea why? Maybe we should use `console` or `shell`?

**Review by giograno — APPROVED** [review]
Kudos for the tutorial 💯 
I have only one question (I don't have a strong opinion about this, but it just came to mind). Since this is a tutorial on collaborative debugging, do you think adding the s3 and ORAS remotes is a bit too much? It's definitively good content, but I am wondering if it makes sense to separate it.

**Review by whummer — COMMENTED** [review]
Kudos for this tutorial @tinyg210 - great to see that we're adding some stories and explicit guidance for how users can address collaborative debugging with pods! 🎉 

I have some comments and suggestions - high level thoughts/summary:
* I generally find the narrative writing style (painting a storyline for the scenario, being a bit witty in a few places, adding a "personal touch" to the text) quite refreshing, but at the same time it can be challenging to stay concise and specific, making it harder to follow. I guess adopting a slightly more neutral and factual writing style could help making the content a bit easier to consume (also, it is hard to make assumptions about the level of maturity/experience of the audience - some readers may understand the context of the more witty parts, others may not)
* To focus more on the collaboration part, would it make sense to include a small figure that illustrates the actual personas involved in the scenario? This would allow us to give them explicit names, and consistently refer to the specific person/persona throughout the document. (that could make it easier to follow, rather than referring to different colleagues, team members, etc)
* Maybe we could slightly expand the scenario with more usage of pods, for example the team member helping with debugging could create the fix, push the cloud pod with the same name (to create version 2), which the other team member can then check out to reproduce the fix (this could nicely illustrate the concept of versioning). Also, stretch goal - can we think of any third role/persona that could come into play here? (Don't have an explicit example in mind, but 3 is often a good number :) )
* Agree with @giograno regarding the sections about alternative cloud pod remotes. Maybe mentioning the S3 remote is helpful, but adding ORAS as well definitely seems like a bit of "overkill". This is generally good content (!) - but could we maybe move this somewhere else in our docs? This would allow us to keep the focus on the collaboration & debugging parts in this tutorial. (We can still mention in a paragraph that alternative storage options are available to enterprise customers, but maybe no need to dedicate entire sections to it in here.)

**Comment by tinyg210** [comment]
Hey @whummer and @giograno, thanks for the feedback! This is definitely valuable input, and here's my plan to address your comments:
- I will remove the last part related to S3 storage and ORAS and reuse the content for a stack byte; in this article, I will only mention the extra possibilities + link to the docs for more info.
- I'll make the writing more clear, with a more neutral tone, to make it easier to read (plus all the other suggestions); I will, however, keep the story, as this is a pretty specific use case.
- I'll introduce a diagram to illustrate the flow between Alice and Bob for the collaboration.
- After Bob experiences the misconfiguration, Alice can use the AWS CLI to fix the missing action in the policy and save the cloud pod (creating v2). Bob will pull the latest version of the cloud pod, use the AWS CLI to view the updated policy, and then make the fix in the Terraform config file. So Bob will push his own fix.
- A third person, Carol, can pull the fixed cloud pod and write integration tests (Bob is only in charge of his own unit tests).

Btw, my initial thought was to have a bug related to bad configuration for the API-Lambda integration (using HTTP GET, instead of POST). At that time, this was not working as expected in LS, but it has since been fixed (https://github.com/localstack/localstack/issues/10473) hehe. I still think the IAM policy enforcement is more important to highlight though.

Please let me know if you'd like to add/remove anything from the new approach (or if I missed anything). Otherwise, I'll be back soon with some fixes.

File: content/en/tutorials/cloud-pods-collaborative-debugging/index.md [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+- [Terraform](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli) or [OpenTofu](https://opentofu.org/docs/intro/install/) and [terraform-local](https://docs.localstack.cloud/user-guide/integrations/terraform/#install-the-tflocal-wrapper-script)
+- Optional for Lambda build & editing: [Maven 3.9.4](https://maven.apache.org/install.html) & [Java 21](https://www.java.com/en/download/help/download_options.html)
+
+### Nice to have
+
+- Basic knowledge of AWS services (API Gateway, Lambda, DynamoDB, IAM)
+- Basic understanding of Terraform for provisioning AWS resources
+
+So let’s say a new colleague joins, clones the application repository, and starts working on the Lambda code. They will add the necessary 
+resources in the Terraform configuration file and some IAM policies that the functions need to be allowed to access the database (after all,
```
[the remark this answers is not in the document]
Comment (tinyg210) [reply content/en/tutorials/cloud-pods-collaborative-debugging/index.md:58]: My thought was to mention this to highlight the granularity of IAM policy enforcement. This is the opposite of `Action = ["dynamodb: *"]`. I'm sure there are apps out there that go to production with terrible security policies 😄
I guess it sounded better in my head and it needs some elaborating.

File: content/en/tutorials/cloud-pods-collaborative-debugging/index.md [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+
+### Other Remote Options
+
+For organizations with specific data regulations, LocalStack offers multiple remote storage options for Cloud Pods,
+allowing full control with on-premises storage if needed.
+That way, Bob, Alice and Carol could collaborate using either an S3 bucket remote storage or an ORAS (OCI Registry as Storage) remote storage.
+The Cloud Pods command-line interface enables users to manage these remotes with ease, by following the instructions in the
+[documentation](https://docs.localstack.cloud/user-guide/state-management/cloud-pods/#remotes).
+
+## Conclusions
```
Comment (HarshCasper) [inline content/en/tutorials/cloud-pods-collaborative-debugging/index.md:322]:
```suggestion
## Conclusion
```

File: content/en/tutorials/cloud-pods-collaborative-debugging/index.md [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+
+- [LocalStack CLI](https://docs.localstack.cloud/getting-started/installation/#localstack-cli) (preferably using `pip`)
+- [Docker](https://docs.docker.com/engine/install/)
+- [Terraform](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli) or [OpenTofu](https://opentofu.org/docs/intro/install/) and [terraform-local](https://docs.localstack.cloud/user-guide/integrations/terraform/#install-the-tflocal-wrapper-script)
+- Optional for Lambda build & editing: [Maven 3.9.4](https://maven.apache.org/install.html) & [Java 21](https://www.java.com/en/download/help/download_options.html)
+
+### Nice to have
+
+- Basic knowledge of AWS services (API Gateway, Lambda, DynamoDB, IAM)
+- Basic understanding of Terraform for provisioning AWS resources
```
Comment (HarshCasper) [inline content/en/tutorials/cloud-pods-collaborative-debugging/index.md:53]: Can we add this to the previous section itself?

File: content/en/tutorials/cloud-pods-collaborative-debugging/index.md [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+
+### Note
+
+This demo application is suitable for AWS and behaves the same as on LocalStack. You can try this out by running the Terraform configuration file against the AWS platform.
+
+![Application Diagram](cloud-pod-collab.png)
+
+### Starting LocalStack
+
+In the root directory, there is a `docker-compose.yml` file that will spin up version 3.3.0 of LocalStack, with an
```
Comment (HarshCasper) [inline content/en/tutorials/cloud-pods-collaborative-debugging/index.md:74]: Any reason for picking a specific version?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: content/en/tutorials/cloud-pods-collaborative-debugging/index.md (hunk 5 above)
Comment (tinyg210) [reply content/en/tutorials/cloud-pods-collaborative-debugging/index.md:74]: it was the latest version at that time. if cloud pods change for future versions i won't go back to change this material

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**